zipkin-lens/src/components/UiConfig/UiConfig.test.jsx

```diff
@@ -35,39 +37,32 @@ const renderUiConfig = async () => {
 };
 
 describe('<UiConfig />', () => {
-  it('fetches config and suspends', async () => {
-    const configPromise = new Promise(() => undefined);
-    const { UI_CONFIG } = await import('../../constants/api');
-    fetchMock.once(UI_CONFIG, configPromise, { overwriteRoutes: true });
-
-    await renderUiConfig();
-    expect(screen.getAllByText('Suspended')).length(1);
-
-    fetchMock.called(UI_CONFIG);
-  });
-
-  it('provides config when resolved', (context) => async () => {
+  it('fetches config, suspends until response, renders after response', async () => {
     const config = { defaultLookback: 100 };
-    const { defaultConfig } = await import('./UiConfig');
     Object.keys(defaultConfig).forEach((key) => {
       config[key] = config[key] || defaultConfig[key];
     });
-    const { UI_CONFIG } = await import('../../constants/api');
-    fetchMock.once(UI_CONFIG, config, { overwriteRoutes: true });
+    const configJson = JSON.stringify(config);
 
-    // TODO: adrian needs help, as this broke when porting to vitest
-    context.skip();
+    let resolve;
+    const configPromise = new Promise((r) => {
+      resolve = r;
+      return undefined;
+    });
+    const fetchSpy = vi
+      .spyOn(global, 'fetch')
+      .mockImplementationOnce(() => configPromise);
 
-    const { rerender } = await renderUiConfig();
+    await renderUiConfig();
     expect(screen.getAllByText('Suspended')).length(1);
 
+    expect(fetchSpy).toHaveBeenCalledWith(UI_CONFIG);
+
+    resolve(new Response(configJson));
     // We need to get off the processing loop to allow the promise to complete and resolve the
     // config.
     await new Promise((resolve) => setTimeout(resolve, 1));
 
-    rerender(); // was rerender(<UiConfig />);
```

Comment:
Presumably I was calling rerender before for some reason, but whether a dep update or what, it seems to not be needed anymore and actually cause the test to fail